FirebaseTestProject/Models/Post.swift

```diff
@@ -22,9 +23,38 @@ struct Post: FirebaseConvertable {
         self.id = UUID()
         self.timestamp = Date()
     }
+    
+    private enum CodingKeys: String, CodingKey {
+        case title
+        case author
+        case text
+        case id
+        case timestamp
+    }
+    
+    //Custom Decoder for `isFavorite` since we don't set it using th Firebase.posts collections
+    init(from decoder: Decoder) throws {
+        let container = try decoder.container(keyedBy: CodingKeys.self)
+        title = try container.decode(String.self, forKey: .title)
+        author = try container.decode(String.self, forKey: .author)
+        text = try container.decode(String.self, forKey: .text)
+        id = try container.decode(UUID.self, forKey: .id)
+        timestamp = try container.decode(Date.self, forKey: .timestamp)
+    }
+    
     static let testPost = Post(title: "Title", text: "Content", author: "First Last")
 }
 
+struct Favorite: FirebaseConvertable {
```

Comment:
I considered the same idea. But, **_theoretically_**, since a user can only favorite a post at most once, the combination of userid+postid intuitively creates a unique value in itself. For example, if we try to reference Firebase for a favorite posts, we will filter the Favorites collection query to only include where userid = currentuser, all returned posts will be a unique `Post`.

Reply:
LMK: It would be easier to follow and understand with a separate `id` but I just considered this route and thought it to be less clunky.

Reply:
Fair enough! If you’d like to keep it this way, I would just make sure that the `favorite(_:)` and `unfavorite(_:)` methods on the `PostService` are acting with this assumption in mind. I added some additional notes below.

Reply:
I see; definitely missing the `unfavorite` filter for userid having no way to "check" it :) Thanks for this!